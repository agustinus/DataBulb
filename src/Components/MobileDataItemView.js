import React from 'react';
import {StyleSheet, View, Text, TouchableOpacity, Image} from 'react-native';
import PropTypes from 'prop-types';

export default class MobileDataItemView extends React.Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    year: PropTypes.string.isRequired,
    total: PropTypes.string.isRequired,
    showIcon: PropTypes.bool,
    onIconClicked: PropTypes.func,
    style: PropTypes.object,
  };

  render() {
    return (
      <View style={{...styles.box, ...this.props.style}}>
        <View style={styles.contentWrapper}>
          <View style={styles.yearWrapper}>
            <Text style={styles.year}>{this.props.year}</Text>
          </View>
          <View style={styles.totalWrapper}>
            <Text style={styles.text}>{'Total Volume of\nMobile Data'}</Text>
            <Text style={styles.total}>{this.props.total}</Text>
          </View>
          {this.props.showIcon ? (
            <TouchableOpacity
              style={styles.icon}
              onPress={this.props.onIconClicked}>
              <Image
                style={styles.image}
                source={require('../res/images/icon-decreased.png')}
              />
            </TouchableOpacity>
          ) : (
            <View style={styles.icon} />
          )}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  box: {
    marginHorizontal: 5,
    backgroundColor: '#ffffff',
    borderRadius: 8,
    height: 90,
    paddingLeft: 15,
    shadowColor: '#000000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 2,
    elevation: 4,
  },
  contentWrapper: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  yearWrapper: {
    flex: 1.5,
    justifyContent: 'center',
  },
  year: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  totalWrapper: {
    flex: 3.5,
    justifyContent: 'flex-end',
  },
  total: {
    fontSize: 20,
    fontWeight: 'normal',
    textAlign: 'right',
  },
  text: {
    fontSize: 12,
    fontWeight: 'normal',
    fontStyle: 'italic',
    textAlign: 'right',
    paddingBottom: 8,
    color: '#606060',
  },
  icon: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    paddingRight: 20,
    height: 25,
    width: 25,
  },
  image: {
    height: 25,
    width: 25,
  },
  loadingContainer: {
    marginTop: 80,
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F5FCFF88',
  },
});
