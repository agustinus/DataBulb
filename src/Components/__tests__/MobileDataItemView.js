import React from 'react';
import MobileDataItemView from '../MobileDataItemView';

import renderer from 'react-test-renderer';

describe('MobileDataItemView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<MobileDataItemView year={'2020'} total={'9.1234567'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly if it's showing the icon", () => {
    const tree = renderer
      .create(
        <MobileDataItemView
          year={'2020'}
          total={'9.1234567'}
          showIcon={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
