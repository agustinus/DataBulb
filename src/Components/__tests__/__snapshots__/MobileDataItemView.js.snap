// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileDataItemView renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "elevation": 4,
      "height": 90,
      "marginHorizontal": 5,
      "paddingLeft": 15,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1.5,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        2020
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3.5,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#606060",
            "fontSize": 12,
            "fontStyle": "italic",
            "fontWeight": "normal",
            "paddingBottom": 8,
            "textAlign": "right",
          }
        }
      >
        Total Volume of
Mobile Data
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "normal",
            "textAlign": "right",
          }
        }
      >
        9.1234567
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "height": 25,
          "justifyContent": "center",
          "paddingRight": 20,
          "width": 25,
        }
      }
    />
  </View>
</View>
`;

exports[`MobileDataItemView renders correctly if it's showing the icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "elevation": 4,
      "height": 90,
      "marginHorizontal": 5,
      "paddingLeft": 15,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1.5,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        2020
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3.5,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#606060",
            "fontSize": 12,
            "fontStyle": "italic",
            "fontWeight": "normal",
            "paddingBottom": 8,
            "textAlign": "right",
          }
        }
      >
        Total Volume of
Mobile Data
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "normal",
            "textAlign": "right",
          }
        }
      >
        9.1234567
      </Text>
    </View>
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "height": 25,
          "justifyContent": "center",
          "opacity": 1,
          "paddingRight": 20,
          "width": 25,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/images/icon-decreased.png",
          }
        }
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
